lib/project_types/script/forms/connect.rb

```diff
@@ -0,0 +1,104 @@
+        end
+
+        orgs = ShopifyCLI::PartnersAPI::Organizations.fetch_with_app(ctx)
+        if orgs.count == 1
+          default = orgs.first
+          ctx.puts(ctx.message("script.application.ensure_env.organization", default["businessName"],
+            default["id"]))
+          default
+        elsif orgs.count > 0
+          CLI::UI::Prompt.ask(ctx.message("script.application.ensure_env.organization_select")) do |handler|
+            orgs.each do |org|
+              handler.option("#{org["businessName"]} (#{org["id"]})") { org }
+            end
+          end
+        else
+          raise Errors::NoExistingOrganizationsError
+        end
+      end
+
+      def stubbed_org
+        {
+          "apps" => [
+            {
+              "appType" => "custom",
+              "apiKey" => "stubbed-api-key",
+              "apiSecretKeys" => [{ "secret" => "stubbed-api-secret" }],
+              "title" => "Fake App (Not connected to Partners)",
+            },
+          ],
+        }
+      end
+
+      def partner_proxy_bypass
+        !ENV["BYPASS_PARTNERS_PROXY"].nil?
+      end
+
+      # copied from lib/shopify_cli/task.rb
+      # should probably just invoke it directly from there
+      def wants_to_run_against_shopify_org?
+        ShopifyCLI::Task.new.wants_to_run_against_shopify_org?(ctx: @ctx)
+        # @ctx.puts(@ctx.message("core.tasks.select_org_and_shop.identified_as_shopify"))
+        # message = @ctx.message("core.tasks.select_org_and_shop.first_party")
+        # CLI::UI::Prompt.confirm(message, default: false)
+      end
+
+      def ask_app(apps)
+        unless ShopifyCLI::Shopifolk.acting_as_shopify_organization?
+          apps = apps.select { |app| app["appType"] == "custom" }
+        end
+
+        if apps.count == 1
+          default = apps.first
+          ctx.puts(ctx.message("script.application.ensure_env.app", default["title"]))
+          default
+        elsif apps.count > 0
+          CLI::UI::Prompt.ask(ctx.message("script.application.ensure_env.app_select")) do |handler|
+            apps.each do |app|
+              handler.option(app["title"]) { app }
+            end
+          end
+        else
+          raise Errors::NoExistingAppsError
+        end
+      end
+
+      def ask_script_uuid(app, extension_point_type)
+        script_service = Layers::Infrastructure::ServiceLocator.script_service(ctx: ctx, api_key: app["apiKey"])
+        scripts = script_service.get_app_scripts(extension_point_type: extension_point_type)
+
+        return nil unless scripts.count > 0 &&
+          CLI::UI::Prompt.confirm(ctx.message("script.application.ensure_env.ask_connect_to_existing_script"))
+
+        CLI::UI::Prompt.ask(ctx.message("script.application.ensure_env.ask_which_script_to_connect_to")) do |handler|
+          scripts.each do |script|
+            handler.option("#{script["title"]} (#{script["uuid"]})") { script["uuid"] }
+          end
+        end
+      end
+    end
+  end
+end
```

Comment:
I think the organization of the code here should be improved.  I think the DDD layers design was not preserved here.  I can try to explain it a bit here.  I tried to create a rough sequence diagram of this code [here](https://drive.google.com/file/d/1i71druxJ1vlrn39hWleHbhAIqFOFlwQQ/view?usp=sharing).  On the bottom, I drew out the layer structure we intended based on DDD.  You can think of the network layers, where each layer has its own set of responsibilities, and communications between layers are very much unidirectional.  We documented how we designed the CLI based on the layers [here](https://docs.google.com/document/d/1s_lVJ1iO_zq8mjkgW435dRij-KXUj4HAwbFBi2_7TbY/edit#heading=h.g9rthrsmuobk).  Because the rest of the CLI isn't structured based on our thinking, sometimes it's a bit harder and we have to encapsulate their logic into the right layers.  

As you can see this class belongs to the UI, where we intend to interact with the user (example: prompt based on some options, or display messages and spinners).  There are definitely too much logic that belongs more into the Application layer than here.  Also, if you try to describe the responsibility of this class, you should realize it's doing way too much.  So, breaking it up, and moving some logic to the Application layer might make sense.  Here is my suggestions, (but interested to hear from others too):
1. Break this into several forms: ask_org, ask_app, ask_script_uuid.
2. ask_org: prompts for org information, and remove the logic in line 20 (this belongs to App layer).
3. ask_app: basically the existing code in this method, but remove the logic on line 70 (belongs to app layer)
4. ask_script_uuid: the calling of SS belongs to the app layer.
5. This might mean that the command::Connect calls direction Application::ConnectApp, which then contains the relevant logic to access the infra layer and calls ask_*.